src/files-action.js

```diff
@@ -20,25 +21,47 @@
  *
  */
 import { generateUrl } from '@nextcloud/router'
+import { translate as t } from '@nextcloud/l10n'
+import { registerFileAction, FileAction, Permission, DefaultType } from '@nextcloud/files'
+/* eslint-disable-next-line import/no-unresolved */
+import ContactSvg from '@mdi/svg/svg/account-multiple.svg?raw'
 
 const mime = 'text/vcard'
 const name = 'contacts-import'
+const nextcloudVersionIsGreaterThanOr28 = parseInt(OC.config.version.split('.')[0]) >= 28
 
-window.addEventListener('DOMContentLoaded', () => {
-	if (OCA.Files && OCA.Files.fileActions) {
-		OCA.Files.fileActions.registerAction({
-			name,
-			displayName: t('contacts', 'Import'),
-			mime,
-			permissions: OC.PERMISSION_READ,
-			iconClass: 'icon-contacts-dark',
-			actionHandler(fileName, context) {
-				const absPath = `${context.dir === '/' ? '' : context.dir}/${fileName}`
-				window.location = generateUrl(`/apps/contacts/import?file=${absPath}`)
-			},
-		})
-		OCA.Files.fileActions.setDefault(mime, name)
-		return
-	}
-	console.error('Unable to register vcf import action')
-})
+if (nextcloudVersionIsGreaterThanOr28) {
```

Comment:
I am not even sure this is needed. Worse case this will register an unused action :shrug: